### PR TITLE
Release snapshots

### LINKED ISSUE #6079: Create a snapshot test for check versions
To ensure that changes to check-versions are not breaking semantics and backwards compatability, we should add snapshots between static versions of the core contracts and assert that the report stays the same. If changes are indeed correct, updating the snapshot should do. It's the equivalent of https://jestjs.io/docs/en/snapshot-testing

Notes:
- Should consider the changes in https://github.com/celo-org/celo-monorepo/pull/6937

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Add CI step `protocol-test-release-snapshots` which compares generated `check-versions` reports to snapshotted version of generated reports
- Requires developer to manually acknowledge version tooling changes by updating the snapshots

### Other changes

- Adds build caching to `scripts/bash/release-lib.sh` for not rebuilding release tags if they are already built

### Tested

Tested in CI

### Related issues

- Fixes #6079

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/protocol/scripts/bash/release-snapshots.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+NETWORK="baklava"
+
+for i in 1 2 3
+do
+	yarn check-versions \
+        -a "celo-core-contracts-v$(($i - 1)).$NETWORK" \
+        -b "celo-core-contracts-v$i.$NETWORK" \
+        -r "releaseData/versionReports/release$i-report.json"
+    # yarn make-release -d -n "$NETWORK" \
```
Comment (kevjue) [inline packages/protocol/scripts/bash/release-snapshots.sh]: Can this commented command be removed?

**Review by kevjue — COMMENTED** [review]
Are the reports in `packages/protocol` needed?  I don't see them being used anywhere.

**Comment by yorhodes** [comment]
> Are the reports in `packages/protocol` needed? I don't see them being used anywhere.

They are used in the CI test
https://github.com/celo-org/celo-monorepo/blob/9cf1a655f5a2b69d63d773dcb911e1938898010b/.circleci/config.yml#L295

**Comment by kevjue** [comment]
> > Are the reports in `packages/protocol` needed? I don't see them being used anywhere.
> 
> They are used in the CI test
> https://github.com/celo-org/celo-monorepo/blob/9cf1a655f5a2b69d63d773dcb911e1938898010b/.circleci/config.yml#L295

I meant the files `packages/protocol/release1-report.json`, `packages/protocol/release2-report.json`, etc...

E.g. I see that `release-snapshots.sh` creates reports in the `releaseData/versionReports/` dir, and those generated reports are git diff'ed in CI.  But I'm still not seeing where those aforementioned files are used.

**Comment by yorhodes** [comment]
> I meant the files `packages/protocol/release1-report.json`, `packages/protocol/release2-report.json`, etc...

Oh my bad will remove

**Review by kevjue — APPROVED** [review]

**Pull request merged**